crates/walrus-service/src/node/db_options.rs

```diff
@@ -0,0 +1,402 @@
+                    shard_sync_progress_column_family_name,
+                },
+            },
+        };
+
+        match self {
+            Self::NodeStatus => node_status_cf_name().to_string(),
+            Self::Metadata => metadata_cf_name().to_string(),
+            Self::AggregateBlobInfo => aggregate_blob_info_cf_name().to_string(),
+            Self::PerObjectBlobInfo => per_object_blob_info_cf_name().to_string(),
+            Self::EventCursor => event_cursor_cf_name().to_string(),
+            Self::PrimarySliver(shard) => primary_slivers_column_family_name(ShardIndex(*shard)),
+            Self::SecondarySliver(shard) => {
+                secondary_slivers_column_family_name(ShardIndex(*shard))
+            }
+            Self::ShardStatus(shard) => shard_status_column_family_name(ShardIndex(*shard)),
+            Self::ShardSyncProgress(shard) => {
+                shard_sync_progress_column_family_name(ShardIndex(*shard))
+            }
+            Self::PendingRecoverSlivers(shard) => {
+                pending_recover_slivers_column_family_name(ShardIndex(*shard))
+            }
+            Self::EventStore => event_processor_constants::EVENT_STORE.to_string(),
+            Self::InitState => event_processor_constants::INIT_STATE.to_string(),
+            Self::Certified => certified_cf_name().to_string(),
+            Self::Attested => attested_cf_name().to_string(),
+            Self::Pending => pending_cf_name().to_string(),
+            Self::FailedToAttest => failed_to_attest_cf_name().to_string(),
+            Self::Custom(name) => name.clone(),
+        }
+    }
+
+    /// Get the RocksDB options for this column family, including any merge operators.
+    pub fn options(&self, db_config: &DatabaseConfig) -> Options {
+        use crate::node::storage::{
+            blob_info::{blob_info_cf_options, per_object_blob_info_cf_options},
+            event_cursor_table_options,
+            metadata_options,
+            node_status_options,
+        };
+
+        match self {
+            Self::NodeStatus => node_status_options(db_config),
+            Self::Metadata => metadata_options(db_config),
+            Self::AggregateBlobInfo => blob_info_cf_options(db_config),
+            Self::PerObjectBlobInfo => per_object_blob_info_cf_options(db_config),
+            Self::EventCursor => {
+                // Use the EventCursorTable::options which includes the proper merge operator.
+                let (_, opts) = event_cursor_table_options(db_config);
+                opts
+            }
+            Self::PrimarySliver(_) | Self::SecondarySliver(_) | Self::PendingRecoverSlivers(_) => {
+                // Sliver column families use the same shard options.
+                db_config.shard().to_options()
+            }
+            Self::ShardStatus(_) | Self::ShardSyncProgress(_) => {
+                // These don't have special options defined.
+                let mut opts = Options::default();
```

Comment:
Also, instead of `Options::default()` i wonder if we need to use `Options::from(&db_config.global())` everywhere